gems/aws-sdk-core/lib/aws-sdk-core/lru_cache.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Aws
+  # @api private
+  # A simple thread safe LRU cache
+  class LRUCache
+    def initialize(options = {})
+      @max_entries = options[:max_entries] || 100
+      @expiration = options[:expiration]
```

Comment:
I know its api private, but some documentation on the options would still help I think (eg, what is expiration? a time? number of seconds?)